subprojects/GOPATH/src/github.com/microsoft/hivedscheduler/pkg/algorithm/cell.go

```diff
@@ -82,28 +103,85 @@ func (c *GenericCell) SetChildren(children CellList) {
 	c.children = children
 }
 
+func (c *GenericCell) AtOrHigherThanNode() bool {
+	return c.atOrHigherThanNode
+}
+
+func (c *GenericCell) IncreaseFreeGpuNum(delta int32) {
+	c.freeGpuNum += delta
+}
+
+func (c *GenericCell) GetFreeGpuNumForPriority(p CellPriority) int32 {
+	return c.freeGpuNumForPriority
+}
+
+func (c *GenericCell) GetUsedGpuNumAtPriority(p CellPriority) int32 {
+	return c.usedGpuNumAtPriority[p]
+}
+
+func (c *GenericCell) IncreaseUsedGpuNumAtPriority(p CellPriority, delta int32) {
+	c.usedGpuNumAtPriority[p] += delta
```

Comment:
>usedGpuNumAtPriority[p] [](start = 3, length = 23)

What if usedGpuNumAtPriority[p] does not exist

Reply:
It will be equivalent to usedGpuNumAtPriority[p] = delta (empty value considered as 0)

---
In reply to: [332820927](https://github.com/microsoft/pai/pull/3684#discussion_r332820927) [](ancestors = 332820927)